src/main/java/graphql/execution/AbsoluteGraphQLError.java

```diff
@@ -0,0 +1,77 @@
+package graphql.execution;
+
+import graphql.ErrorType;
+import graphql.GraphQLError;
+import graphql.language.SourceLocation;
+import graphql.schema.DataFetcher;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * A {@link GraphQLError} that has been changed from a {@link DataFetcher} relative error to an absolute one.
+ */
+class AbsoluteGraphQLError implements GraphQLError {
+
+    private final List<SourceLocation> locations;
+    private final List<Object> absolutePath;
+    private final GraphQLError relativeError;
```

Comment:
Looks like you don't need to keep a reference to the `relativeError`, doesn't seem to be used anywhere but in the constructor.